packages/flutter/test/widgets/raw_menu_anchor_test.dart

```diff
@@ -0,0 +1,5787 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// This file is run as part of a reduced test set in CI on Mac and Windows
+// machines.
+@Tags(<String>['reduced-test-set'])
+library;
+
+import 'dart:ui' as ui;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+// Tests that apply to select constructors have a suffix that indicates which
+// constructor the test applies to:
+//  * [Default]: Applies to [RawMenuAnchor],
+//  * [OverlayBuilder]: Applies to [RawMenuAnchor.overlayBuilder]
+//  * [Panel]: Applies to [RawMenuAnchor.menuPanel].
+//  * [Overlays]: Applies to [RawMenuAnchor] and [RawMenuAnchor.overlayBuilder]
+// Otherwise, the test applies to all constructors.
```

Comment:
@dkwingsmt told know it was discouraged, but I'm not sure if that has changed. Groups would definitely make it easier to navigate.

Reply:
The main issue I've run into with groups is that each individual test description tends to be more vague.

Using [**input_decorator_test.dart**](https://github.com/flutter/flutter/blob/master/packages/flutter/test/material/input_decorator_test.dart) as an example:

```dart
group('for filled text field', () {
  group('when field is enabled', () {
    testWidgets('container has correct height and shape', (WidgetTester tester) async {
      // ...
    });
  });
  group('when field is disabled', () {
    testWidgets('container has correct height and shape', (WidgetTester tester) async {
      // ...
    });
  });
})
```

I've gotten into the habit of copy-pasting the test name to find a test, so when there are a bunch of duplicates it becomes more difficult to navigate. I don't believe there's any explicit advice in our documentation about whether to use groups, though [**Make each test entirely self-contained**](https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#make-each-test-entirely-self-contained) seems to go against several things you might want to use groups for.

Reply:
Thanks for pointing out this rule, yes groups also have drawbacks. 
Ideally we don't need them if [Prefer more test files, avoid long test files](https://github.com/flutter/flutter/blob/master/docs/contributing/testing/Writing-Effective-Tests.md) was used more often. I will definitely think about it when I will work again on InputDecorator.

@davidhicks980 Let's keep the comment as-is for the moment, and once this PR is merged, it might be interesting to split the tests into several files.